server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -64,12 +64,14 @@ def deploy
       end
     end
 
-    def validate
-      status = @deployment.valid? ? 200 : 422
-
-      render status: status, json: {
-        :errors => @deployment.errors.full_messages,
-        :warnings => @deployment.warnings
+    def validation
```

Comment:
I liked your other changes to the controller. For one, I think returning 200 in all cases makes sense. I'm happy to make those changes in a separate PR though if you want to get this merged.